src/services/observation-service.ts

```diff
@@ -48,33 +48,31 @@ export const filterFieldByLocale = (key: Locale, lang: Locale): boolean => {
   return lang === key;
 };
 
-const reduceWithCount = (arr: any[], columnName: string) => {
-  if (arr[0].count === '0') {
-    return {};
+export const mapLocale = (obsEntry: [string, any], lang: Locale): [string, any] => {
+  const [observationKey, observationValue] = obsEntry;
+  if (typeof observationValue === 'object' && observationValue !== null) {
+    const value = Object.entries(observationValue)
+      .filter(([field]) => filterFieldByLocale(field as Locale, lang))
+      .map(entrie => (entrie[0] === lang ? ['desc', entrie[1]] : entrie))
+      .reduce((acc, [subfield, subValue]) => Object.assign(acc, { [subfield as string]: subValue as any }), {});
+
+    return [observationKey, value];
   }
-  return {
-    [columnName]: arr.map(row => ({
-      value: row[columnName] === undefined ? { ...row, count: undefined } : row[columnName],
-      count: row.count,
-    })),
-  };
+  return obsEntry;
+};
+
+export const sanitizeUser = (obsEntry: [string, any]): [string, any] => {
+  const [observationKey, observationValue] = obsEntry;
+  if (observationKey === 'finder') {
+    const finder = User.sanitizeUser(observationValue);
+    return [observationKey, finder];
+  }
+  return obsEntry;
 };
 
-const aggregationColumns: string[] = ['speciesMentionedId', 'speciesConcludedId', 'verified', 'ringId'];
 const aggregationSearch: string[] = ['search', 'pageNumber', 'pageSize', 'sortingColumn', 'sortingDirection'];
-const aggregationForeignKeys: ((repository: Repository<Observation>) => Promise<{ [x: string]: any }>)[] = [
-  async repository => {
-    const res = await repository
-      .createQueryBuilder('observation')
-      .select(['finder."id"', 'finder."firstName"', 'finder."lastName"', 'finder."role"', 'count(*)'])
-      .innerJoin('observation.finder', 'finder')
-      .groupBy('finder."id"')
-      .getRawMany();
-    return reduceWithCount(res, 'finder');
-  },
-];
-
-export const parseWhereParams = (query: ObservationAggregations, user: User): FindOneOptions<Observation> => {
+
+export const parseWhereParams = (user: User, query: ObservationAggregations): FindOneOptions<Observation> => {
   const params = Object.entries(query)
     .filter(entrie => !aggregationSearch.includes(entrie[0])) // filter search aggregations, only ObservationAggregations could be applied
```

Comment:
for both `map` and `filter` is preferable to use destructuring to increase readability